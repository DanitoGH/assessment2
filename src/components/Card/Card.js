import React from 'react';
import PropTypes from 'prop-types';

import { Container, Row , Col, Card , FormLabel } from 'react-bootstrap'

import Checkbox from "../Checkbox/Checkbox";
import Dropdown from "../Dropdown/Dropdown";


const RoomsCard = ({ roomNumber, roomChecked, onChecked, adultRoomValue ,childrenRoomValue, onChanged }) => {

    return (
        <Card  style={{ width: '15rem' }}>
            <Card.Body className={roomChecked || roomNumber === 1?"bg-primary text-white" : "bg-secondary  text-white"}  style={{ padding: '0.7rem' }}>
               <Card.Header style={{ padding: '0.4rem' }}>
                  <Checkbox 
                      label={`Room ${roomNumber}`}
                      onChange={onChecked}
                      checked={roomChecked}
                      roomNumber={roomNumber}
                  />
               </Card.Header>
               <Container>
                <Row>
                  <Col>
                   <FormLabel className="mt-2">Adults (18+)</FormLabel>
                    <Dropdown 
                      options={[
                        { value: 1, label: 1 },
                        { value: 2, label: 2 }
                        ]}
                        onChange={onChanged}
                        roomChecked={roomNumber === 1? true : roomChecked}
                        dropdownCategory="Adults"
                        roomNumber={roomNumber}
                        adultRoomValue={adultRoomValue}
                        childrenRoomValue={childrenRoomValue}
                     />
                    </Col>
                    <Col>
                     <FormLabel className="mt-2">Children (0-17)</FormLabel>
                      <Dropdown 
                        options={[
                            { value: 0, label: 0 },
                            { value: 1, label: 1 },
                            { value: 2, label: 2 }
                            ]}
                            onChange={onChanged}
                            roomChecked={roomNumber === 1? true : roomChecked}
                            dropdownCategory="Children"
                            roomNumber={roomNumber}
                            adultRoomValue={adultRoomValue}
                            childrenRoomValue={childrenRoomValue}
                      />
                    </Col>
                </Row>
              </Container>
         </Card.Body>
        </Card>
    );
};

RoomsCard.protoTypes = {
  roomNumber: PropTypes.array.isRequired,
  roomChecked: PropTypes.bool.isRequired,
  onChecked: PropTypes.func.isRequired,
  onChanged: PropTypes.func.isRequired
}

RoomsCard.protoTypes = {
  adultRoomValue: '',
  childrenRoomValue: ''
}

 export default RoomsCard;

