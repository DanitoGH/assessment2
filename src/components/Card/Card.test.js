import { render } from '@testing-library/react';
import { shallow }  from "enzyme";
import toJSON from "enzyme-to-json";

import Card from './Card';


test('properly renders card component', () => {
  render(<Card />);
})

test("matches snapshot", () => {
  const wrapper = shallow(<Card />);

  expect(toJSON(wrapper)).toMatchSnapshot();
})
