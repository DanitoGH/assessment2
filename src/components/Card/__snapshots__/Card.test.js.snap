// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Card
  body={false}
  style={
    Object {
      "width": "15rem",
    }
  }
>
  <CardBody
    className="bg-secondary  text-white"
    style={
      Object {
        "padding": "0.7rem",
      }
    }
  >
    <CardHeader
      style={
        Object {
          "padding": "0.4rem",
        }
      }
    >
      <Checkbox
        label="Room undefined"
      />
    </CardHeader>
    <Container
      fluid={false}
    >
      <Row
        noGutters={false}
      >
        <Col>
          <FormLabel
            className="mt-2"
            column={false}
            srOnly={false}
          >
            Adults (18+)
          </FormLabel>
          <Dropdown
            dropdownCategory="Adults"
            options={
              Array [
                Object {
                  "label": 1,
                  "value": 1,
                },
                Object {
                  "label": 2,
                  "value": 2,
                },
              ]
            }
          />
        </Col>
        <Col>
          <FormLabel
            className="mt-2"
            column={false}
            srOnly={false}
          >
            Children (0-17)
          </FormLabel>
          <Dropdown
            dropdownCategory="Children"
            options={
              Array [
                Object {
                  "label": 0,
                  "value": 0,
                },
                Object {
                  "label": 1,
                  "value": 1,
                },
                Object {
                  "label": 2,
                  "value": 2,
                },
              ]
            }
          />
        </Col>
      </Row>
    </Container>
  </CardBody>
</Card>
`;
